data/typelibrary/utils-1.0.0/typelib/splitting/BYTE_TO_QUARTERS.fct

```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Function Name="BYTE_TO_QUARTERS" Comment="">
+	<Identification Standard="61499-1" Description="Copyright (c) 2024 HR Agrartechnik GmbH    &#10;    &#10;This program and the accompanying materials are made     &#10;available under the terms of the Eclipse Public License 2.0     &#10;which is available at https://www.eclipse.org/legal/epl-2.0/     &#10;     &#10;SPDX-License-Identifier: EPL-2.0" >
+	</Identification>
+	<VersionInfo Version="1.0" Author="franz" Date="2024-02-22">
+	</VersionInfo>
+	<CompilerInfo>
+	</CompilerInfo>
+	<InterfaceList>
+		<EventInputs>
+			<Event Name="REQ" Type="Event" Comment="Normal Execution Request">
+				<With Var="IN"/>
+			</Event>
+		</EventInputs>
+		<EventOutputs>
+			<Event Name="CNF" Type="Event" Comment="Execution Confirmation">
+				<With Var="OUT_00"/>
+				<With Var="OUT_01"/>
+				<With Var="OUT_02"/>
+				<With Var="OUT_03"/>
+			</Event>
+		</EventOutputs>
+		<InputVars>
+			<VarDeclaration Name="IN" Type="BYTE" Comment="BYTE Input"/>
+		</InputVars>
+		<OutputVars>
+			<VarDeclaration Name="OUT_00" Type="BYTE" Comment="QUARTER BYTE 00"/>
+			<VarDeclaration Name="OUT_01" Type="BYTE" Comment="QUARTER BYTE 01"/>
+			<VarDeclaration Name="OUT_02" Type="BYTE" Comment="QUARTER BYTE 02"/>
+			<VarDeclaration Name="OUT_03" Type="BYTE" Comment="QUARTER BYTE 03"/>
+		</OutputVars>
+	</InterfaceList>
+	<FunctionBody>
+		<ST><![CDATA[FUNCTION BYTE_TO_QUARTERS
+
+VAR_INPUT
+	IN : BYTE;    // BYTE Input
+END_VAR
+
+VAR_OUTPUT
+	OUT_00 : BYTE;    // QUARTER BYTE 00
+	OUT_01 : BYTE;    // QUARTER BYTE 01
+	OUT_02 : BYTE;    // QUARTER BYTE 02
+	OUT_03 : BYTE;    // QUARTER BYTE 03
+END_VAR
+
+(* this Function extracts the 4 QUARTER BYTE from a byte *)
+OUT_00 := SHR(IN AND BYTE_QUARTER_00, SHFT_QUARTER_00);
```

Comment:
This can be implemented easier with .%Bn access oeprators

Reply:
@MartinMelikMerkumians i am interested in how this can be done. 
we used it in BYTE_TO_BOOLS. but how to use it for a Quarter Byte, say 2 Bits ?

Reply:
we could use .%Bn for a Byte, and still extract the 2 bits with a AND !

Reply:
Oh sorry, I mixed up the lines. Those 2 bit accesses cant be done easier

Reply:
agree. many others yes.